api/v1alpha1/spire_server_config_types.go

```diff
@@ -0,0 +1,206 @@
+	Spec              SpireServerConfigSpec   `json:"spec,omitempty"`
+	Status            SpireServerConfigStatus `json:"status,omitempty"`
+}
+
+// SpireServerConfigSpec will have specifications for configuration related to the spire server.
+type SpireServerConfigSpec struct {
+
+	// trustDomain to be used for the SPIFFE identifiers
+	// +kubebuilder:validation:Required
+	TrustDomain string `json:"trustDomain,omitempty"`
+
+	// clusterName will have the cluster name required to configure spire server.
+	// +kubebuilder:validation:Required
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// bundleConfigMap is Configmap name for Spire bundle, it sets the trust domain to be used for the SPIFFE identifiers
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=spire-bundle
+	BundleConfigMap string `json:"bundleConfigMap"`
+
+	// JwtIssuer is the JWT issuer domain. Defaults to oidc-discovery.$trustDomain if unset
+	// +kubebuilder:validation:Optional
+	JwtIssuer string `json:"jwtIssuer,omitempty"`
+
+	// spireServerKeyManager has configs for the spire server key manager.
+	// +kubebuilder:validation:Optional
+	SpireServerKeyManager *SpireServerKeyManager `json:"spireServerKeyManager,omitempty"`
+
+	// CASubject contains subject information for the Spire CA.
+	// +kubebuilder:validation:Optional
+	CASubject *CASubject `json:"caSubject,omitempty"`
+
+	// persistence has config for spire server volume related configs
+	// +kubebuilder:validation:Optional
+	Persistence *Persistence `json:"persistence,omitempty"`
+
+	// spireSQLConfig has the config required for the spire server SQL DataStore.
+	// +kubebuilder:validation:Optional
+	Datastore *DataStore `json:"spireSQLConfig,omitempty"`
+
+	// labels to apply to all resources created for operator deployment.
+	// +mapType=granular
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// resources are for defining the resource requirements.
+	// Cannot be updated.
+	// ref: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +kubebuilder:validation:Optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// affinity is for setting scheduling affinity rules.
+	// ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/
+	// +kubebuilder:validation:Optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// tolerations are for setting the pod tolerations.
+	// ref: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+	// +kubebuilder:validation:Optional
+	// +listType=atomic
+	Tolerations []*corev1.Toleration `json:"tolerations,omitempty"`
+
+	// nodeSelector is for defining the scheduling criteria using node labels.
+	// ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +kubebuilder:validation:Optional
+	// +mapType=atomic
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
De-duplicate with an embedded struct:

```suggestion
	ZeroTrustWorkloadIdentityManagerSpec `json:",inline"`
```

Reply:
Added a commonConfig for all APIs.